docs/register/prop-method-injection.rst

```diff
@@ -21,7 +21,118 @@ To support :doc:`circular dependencies <../advanced/circular-dependencies>`, use
 
     builder.Register(c => new A()).OnActivated(e => e.Instance.B = e.Context.Resolve<B>());
 
-If the component is a :ref:`reflection component <register-registration-reflection-components>`, use the ``PropertiesAutowired()`` modifier to inject properties:
+Required Properties
+-------------------
+
+From Autofac 7.0 onwards, for :ref:`reflection components <register-registration-reflection-components>`, all `required properties <https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/required>`_ are automatically resolved at the time of object construction, and are generally treated in much the same way as mandatory constructor arguments.
+
+For example, given the following type:
+
+.. sourcecode:: csharp
+
+    public class MyComponent
+    {
+      public required ILogger Logger { get; set; }
+
+      public required IConfigReader ConfigReader { get; set; }
+    }
+
+When the component is resolved, Autofac will populate the ``Logger`` and ``ConfigReader`` properties as if they were constructor parameters.
```

Comment:
```suggestion

      public IDatabaseContext Context { get; set; }
    }

When the component is resolved, Autofac will populate the ``Logger`` and ``ConfigReader`` properties as if they were constructor parameters. The ``Context`` property will be treated like a standard property and will not be populated by default.
```

I figured it would be valuable to show that just because some props get required, no, that doesn't mean they all suddenly become required. This seems obvious yet is also a question I can easily see someone misunderstanding.